Add .gitignore file and Swagger spec

jsfan opened the pull request:
This PR is meant to be the work relating to the Swagger spec.

As we didn't have any, yet, I've also added a .gitignore file.

Commit:
Added .gitignore file (covers Python, Jetbrains IDEs and VSCode)

Commit:
Added Swagger stub

Ashar67 commented on swagger/swagger.yaml:
```diff
@@ -0,0 +1,299 @@
+  /church:
+    post:
+      tags:
+      - "member"
```
members should not be allowed to add/remove/modify churches right? I can update this on my commit

Ashar67 committed:
Update swagger.yaml

- added Question and Group endpoints
- minor format checks for field types

Ashar67 committed:
Update swagger.yaml

- added church settings

Commit:
Endpoint additions and cleanup

* Increased granularity of different objects (e.g. UserIn, User and PublicUser)
* Changed "invalid input" to code 400 (bad request) which is the correct code
* Added direct manipulation of contact methods (not via user)
* Added user listing for leaders (to be restricted to leader's church members)

Ashar67 commented on swagger/swagger.yaml:
```diff
@@ -0,0 +1,710 @@
+        type: "string"
+        format: "uuid"
+
+  PublicUser:
```
Is the only difference between `PublicUser` and `User` the DOB? (For privacy issues, I'm assuming?)

Ashar67 commented on swagger/swagger.yaml:
```diff
@@ -0,0 +1,710 @@
+    put:
+      tags:
+      - "leader"
+      summary: "Update question"
```
Would it be easier to get rid of `Update question` and just allow delete and add questions?

Ashar67 committed:
Bulletin Addition

- added group message sending and retrieval via "Bulletin"
- slight cleanup of unused parameters

Ashar67 committed:
Restructure Messages

- restructure for cleaner code generation

Commit:
Small fix to user endpoint

Commit:
Merge branch 'gitignore-swagger' of github.com:UpstreamCode/interconnect-backend into gitignore-swagger

Ashar67 committed:
Add to Swagger

-  updated functions

Commit:
Added inheritance to models

Ashar67 committed:
Restructure + Fixes + Firebase

- restructured code to be more logical: all `leader` actions are prefaced with "church" and all `member` actions with "user"
- fixed incorrect references
- added firebase id for login verification

oosawaye committed:
support for bulk email invites

oosawaye committed:
updated invite body format

oosawaye committed:
update yaml to remove some unneeded fields
